tf-modules/distribution/variables.tf

```diff
@@ -1,5 +1,47 @@
 # Required
 
+variable "log_api_gateway_to_cloudwatch" {
+  type        = bool
+  default     = false
+  description = "Boolean switch to enable/disable logging of Api Gateway distribution traffic to CloudWatch."
+}
+
+variable "log_to_shared_destination" {
+  type        = bool
+  default     = false
+  description = "Boolean switch to enable/disable propagation of Api Gateway and s3 Access Logs to a shared destination. If enabled, also use log_to_shared_destination."
+}
+
+variable "log_destination_arn" {
+  type    = string
+  default = null
+  default = "Only used (and must be set) if both log_api_gateway_to_cloudwatch and log_to_shared_destination are true."
+}
+
+variable "s3_replicator_source_bucket" {
+  type        = string
+  default     = null
+  description = "Used with the s3-replicator module. Source bucket from which new objects will be replicated."
+}
+
+variable "s3_replicator_source_prefix" {
+  type        = string
+  default     = null
+  description = "Used with the s3-replicator module. Any new objects in source_bucket that start with this prefix will be replicated."
+}
+
+variable "s3_replicator_target_bucket" {
+  type        = string
+  default     = null
+  description = "Used with the s3-replicator module. Target bucket to which new objects will be replicated."
+}
+
+variable "s3_replicator_target_prefix" {
+  type        = string
+  default     = null
+  description = "Used with the s3-replicator module. New objects will be replicated with this prefix to the target_bucket."
+}
+
```

Comment:
We talked about this a little before, but I'm not sure if that kind of change is compatible with the TEA configuration. @yjpa7145 any thoughts on this?

Reply:
I believe it's the distribution api stage that it needs to match, not the backend api stage.

Looking at https://github.com/asfadmin/thin-egress-app/blob/devel/terraform/variables.tf#L118, there is no default stage being set in the TEA TF module, so it's deferring to the default defined in the TEA CloudFormation template here: https://github.com/asfadmin/thin-egress-app/blob/devel/cloudformation/thin-egress-app.yaml#L149

I think the right way to handle it is to add an output to the CF stack that returns the stage that was actually used. That output should then become an output of the TEA TF module, and then used as the `api_gateway_stage` input variable of the distribution module.

In the short term, since people are most likely going to be setting the`stage_name` input variable of the thin-egress-app TF module in their root TF module, you can probably get away with setting the `api_gateway_stage = module.thin_egress_app.stage_name`. As long as they are setting that value explicitly and not using the default, then it'll work.

As I'm typing, I realize that the solution I described in the previous paragraph is almost certainly a bad idea. I'm hesitant to hold up this ticket and require another release of thin egress, but it certainly seems like it's going to cause people heartache when their API Gateway stages don't match up and they aren't sure why.